big_tests/tests/mod_event_pusher_rabbit_SUITE.erl

```diff
@@ -0,0 +1,321 @@
+%% Test helpers
+%%--------------------------------------------------------------------
+
+-spec listen_to_rabbit() -> proplists:proplist().
+listen_to_rabbit() ->
+    {ok, Connection} =
+        amqp_connection:start(#amqp_params_network{host = "localhost"}),
+    {ok, Channel} = amqp_connection:open_channel(Connection),
+    [{rabbit_connection, Connection}, {rabbit_channel, Channel}].
+
+-spec close_rabbit_connection(Config :: proplists:proplist()) -> ok | term().
+close_rabbit_connection(Config) ->
+    Connection = proplists:get_value(rabbit_connection, Config),
+    Channel = proplists:get_value(rabbit_channel, Config),
+    amqp_channel:close(Channel),
+    amqp_connection:close(Connection).
+
+-spec listen_to_presence_events_from_rabbit(JIDs :: [binary()],
+                                            Config :: proplists:proplist()) ->
+    ok | term().
+listen_to_presence_events_from_rabbit(JIDs, Config) ->
+    Connection = proplists:get_value(rabbit_connection, Config),
+    Channel = proplists:get_value(rabbit_channel, Config),
+    Queue = declare_rabbit_queue(Channel),
+    QueueBindings = presence_bindings(Queue, JIDs),
+    wait_for_exchanges_to_be_created(Connection, JIDs),
+    bind_queues_to_exchanges(Channel, QueueBindings),
+    subscribe_to_rabbit_queue(Channel, Queue).
+
+-spec wait_for_exchanges_to_be_created(Connection :: pid(),
+                                       Exchanges :: [binary()]) -> pid().
+wait_for_exchanges_to_be_created(Connection, Exchanges) ->
+    {ok, TestChannel} = amqp_connection:open_channel(Connection),
+    [is_exchange_present(Connection, TestChannel, Exchange,
+                         ?IF_EXCHANGE_EXISTS_RETRIES)
+     || Exchange <- Exchanges],
+    ok.
+
+-spec declare_rabbit_queue(Channel :: pid()) -> binary().
+declare_rabbit_queue(Channel) ->
+    #'queue.declare_ok'{queue = Queue} =
+        amqp_channel:call(Channel, #'queue.declare'{exclusive = true}),
+    Queue.
+
+-spec presence_bindings(Queue :: binary(), JIDs :: [binary()]) ->
+    [rabbit_binding()].
+presence_bindings(Queue, JIDs) ->
+    [{Queue, JID, presence_rk(JID)} || JID <- JIDs].
+
+-spec bind_queues_to_exchanges(Channel :: pid(),
+                               Bindings :: [rabbit_binding()]) ->
+    [amqp_client:amqp_method() | ok | blocked | closing].
+bind_queues_to_exchanges(Channel, Bindings) ->
+    [bind_queue_to_exchange(Channel, Binding) || Binding <- Bindings].
+
+bind_queue_to_exchange(Channel, {Queue, Exchange, RoutingKey}) ->
+    amqp_channel:call(Channel, #'queue.bind'{exchange = Exchange,
+                                             routing_key = RoutingKey,
+                                             queue = Queue}).
+
+-spec is_exchange_present(Connection :: pid(), Channel :: pid(),
+                                      Exchange :: binary(),
+                                      Retries :: non_neg_integer()) ->
+    true | no_return().
+is_exchange_present(_Connection, Channel, Exchange, 0) ->
```

Comment:
If you count down to `0`, you have more runs, than specified in `Retries` :P . You should countdown to `1`